File: bailoteca-backend/src/main/java/com/bailoteca/models/dtos/UsuarioRequest.java
package com.bailoteca.models.dtos;

import com.bailoteca.models.enums.Rol;
import lombok.Data;
import java.time.LocalDate;

/**
 * DTO utilizado para registrar un nuevo usuario en el sistema.
 * Este DTO expone solo los campos necesarios desde el frontend
 * y permite mantener la entidad aislada.
 */

@Data
public class UsuarioRequest {
    private String nombre;
    private String apellido;
    private String correo;
    private String contrasenna;
    private Rol rol = Rol.USUARIO;
    private String telefono;
    private String direccion;
    private LocalDate fechaNacimiento;
}

File: bailoteca-frontend/app/src/main/java/com/example/bailotecaapp/model/dto/UsuarioRequest.kt
package com.example.bailotecaapp.model.dto

import com.example.bailotecaapp.model.enums.Rol
import kotlinx.serialization.Serializable

@Serializable
data class UsuarioRequest(
    val nombre: String,
    val apellido: String,
    val correo: String,
    val contrasenna: String,
    val rol: Rol,
    val telefono: String = "",
    val direccion: String = "",
    val fechaNacimiento: String? = null //yyyy-MM-dd
)